tests/pyreverse/test_writer.py

```diff
@@ -257,3 +257,84 @@ def test_package_name_with_slash(default_config: PyreverseConfig) -> None:
     writer.write([obj])
 
     assert os.path.exists("test_package_name_with_slash_.dot")
+
+
+def test_should_show_node_no_depth_limit(default_config: PyreverseConfig) -> None:
+    """Test that nodes are shown when no depth limit is set."""
+    writer = DiagramWriter(default_config)
+    writer.max_depth = None
+
+    assert writer.should_show_node("pkg")
+    assert writer.should_show_node("pkg.subpkg")
+    assert writer.should_show_node("pkg.subpkg.module")
+    assert writer.should_show_node("pkg.subpkg.module.submodule")
+
+
+@pytest.mark.parametrize("max_depth", range(5))
+def test_should_show_node_with_depth_limit(
+    default_config: PyreverseConfig, max_depth: int
+) -> None:
+    """Test that nodes are filtered correctly when depth limit is set.
+
+    Depth counting is zero-based, determined by number of dots in path:
+    - pkg                   -> depth 0 (0 dots)
+    - pkg.subpkg           -> depth 1 (1 dot)
+    - pkg.subpkg.module    -> depth 2 (2 dots)
+    - pkg.subpkg.module.submodule -> depth 3 (3 dots)
```

Comment:
```suggestion
    - 'pkg'                  -> depth 0 (0 dots)
    - 'pkg.subpkg'           -> depth 1 (1 dot)
    - 'pkg.subpkg.module'    -> depth 2 (2 dots)
    - 'pkg.subpkg.module.submodule' -> depth 3 (3 dots)
```
Somthings's very wrong with the way the github action display pylint message but I guess this fixes the spelling